crates/apollo-compiler/src/diagnostics.rs

```diff
@@ -269,6 +269,18 @@ pub enum DiagnosticData {
     DisallowedSubselection,
     #[error("interface, union and object types must have a subselection set")]
     MissingSubselection,
+    #[error("`{name}` can not be declared on scalar types")]
+    InvalidFragmentType {
+        /// Name of the fragment
+        name: String,
+        /// Name of the type on which the fragment is declared
+        ty: String,
+    },
+    #[error("`{name}` must be used in schema definition")]
```

Comment:
fragments are not declared on a schema definition, maybe let's change it to this instead?
```suggestion
    #[error("`{name}` must be used in an operation definition")]
```